src/screenfull.js

```diff
@@ -93,13 +93,20 @@
 				// keyboard in fullscreen even though it doesn't.
 				// Browser sniffing, since the alternative with
 				// setTimeout is even worse.
+				let result;
 				if (/ Version\/5\.1(?:\.\d+)? Safari\//.test(navigator.userAgent)) {
-					elem[request]();
+					result = elem[request]();
 				} else {
-					elem[request](keyboardAllowed ? Element.ALLOW_KEYBOARD_INPUT : {});
+					result = elem[request](keyboardAllowed ? Element.ALLOW_KEYBOARD_INPUT : {});
 				}
-
-				this.on('change', onFullScreenEntered);
+				
+				if (!(result instanceof Promise)) {
```

Comment:
This is not the correct way to check if something is a promise. It will not work for users using a Promise polyfill.